### ISSUE TITLE
Face detection and cropping?

### ISSUE THREAD

**Issue by 921j** [mifi/lossless-cut#1768]
### I have a lot of issues to go through, so in order to make it easier for me to help you, I ask that you please try these things first

- [X] Check out all the menus and keyboard shortcuts. Maybe the feature already exists?
- [X] Try with the [newest version from GitHub](https://github.com/mifi/lossless-cut/releases/latest)
- [X] Read the [documentation](https://github.com/mifi/lossless-cut) and [FAQ, Known issues, Troubleshooting](https://github.com/mifi/lossless-cut/blob/master/issues.md) to see if the feature already exists
- [X] See if someone already suggested this under [Issues](https://github.com/mifi/lossless-cut/issues) or [Discussions](https://github.com/mifi/lossless-cut/discussions)

### Description

I don’t know if this request is too much :)
Sometimes I need to cut the clearest part of the person's face into a fragment or screenshot, which consumes a lot of time to find this frame. If LosslessCut can support this feature, it will be the great and unique!

**Comment by 921j** [comment]
Similar to software "Face Crop Jet" and "Face Cropper", they can crop a face avatar from a images to make ID card. I hope to find a lightweight video tool to detect and crop the face avatar from the video, but currently the face detection feature is some complicated and bloated face-changing software can do, like DeepFaceLab, that is deployed hardly, and requires strong hardware support. No one considers to make a tool to auto crop face avatar from a video, maybe it is because development is too difficult and the income is not much?
If Losslescut integrates this function, there is no doubt that it will be wonderful and unique, and this can be used as a payment function for LC's users to choose from.

**Comment by mifi** [comment]
Not sure what's the use case for making ID cards from videos and how common this is. Normally I'd think that pictures are taken at a photo booth or pre-captured photos are used, not taken from a video clip?

**Comment by 921j** [comment]
Of course there are very many application scenarios!
When we edit videos, the most commonly reference core objects are characters, not black frames or white frames.
Whether it is a profile or video file for "strangers", or star the tidbits, the lecture videos of different characters, or making a video embedded cover, it is useful!

**Comment by mifi** [comment]
ah ok, I think I misunderstood - I thought you wanted to auto-find and auto-crop faces in order to make an ID card.
So do I understand it correctly, you want losslesscut to be able to scan through a video (similar to blackdetect) and find all frames that have faces in them, and then create segments for each "sequence of frames with faces". e.g.:
- frame 1 has 1 face
- frame 2 has 1 face
- frame 3 has 2 faces
- frame 4 has 1 face
- frame 5 has 0 faces
- frame 6 has 3 faces
- frame 7 has 1 face

this would result in 2 segments:
- segment 1: frame 1 to 4
- segment 2: frame 6 to 7

**Comment by 921j** [comment]
Yes, just like some professional audio editing software can detect and extract people voice from a song, detect face from a video, and show the waveform to the editor for reference. Different people use waves of different colors, side or positive face represented by wave crest.
I believe that when most people are editing videos, the most often encountered situation is editing around a certain or some characters in the video, doesn't anyone like me, every every times it must spend a lot of time looking for this scenes where these characters appear?

**Comment by mifi** [comment]
Probably have to use something like https://github.com/justadudewhohacks/face-api.js
